File: packages/api-models-ts/src/Roles.ts
/**
 * Request and response models for `/api/roles`.
 *
 * @since 1.0.0
 */

import * as v from "valibot";

import * as Schemas from "./Schemas.ts";

/**
 * A named collection of ACLs (upstream baselayer `Role`).
 *
 * The handler replaces the `acls` relationship with a list of ACL IDs.
 *
 * @since 1.0.0
 * @category Models
 */
export const Role = Schemas.model(
    v.strictObject({
        id: v.string(),
        created_at: Schemas.NullishTimestamp,
        modified: Schemas.NullishTimestamp,
        acls: Schemas.list(v.string()),
    })
);

/**
 * @since 1.0.0
 * @category Models
 */
export type Role = v.InferOutput<typeof Role>;

File: packages/api-models-py/skyportal_py_models/roles.py
"""Request and response models for ``/api/roles``."""

from __future__ import annotations

import datetime

from pydantic import BaseModel, ConfigDict, Field


class Role(BaseModel):
    """A named collection of ACLs (upstream baselayer ``Role``)."""

    # The handler replaces the ``acls`` relationship with a list of ACL IDs.

    model_config = ConfigDict(extra="forbid")

    id: str
    created_at: datetime.datetime | None = None
    modified: datetime.datetime | None = None
    acls: list[str] = Field(default_factory=list)
